File: facade/src/main/java/org/tdf/sunflower/facade/Miner.java
package org.tdf.sunflower.facade;


import org.tdf.common.util.HexBytes;

public interface Miner {
    Miner NONE = new Miner() {
        @Override
        public void start() {

        }

        @Override
        public void stop() {

        }
    };

    void start();

    void stop();

    default HexBytes getMinerAddress() {
        return HexBytes.empty();
    }
}

File: facade/src/main/java/org/tdf/sunflower/facade/Miner.kt
package org.tdf.sunflower.facade

import org.tdf.common.util.Address
import org.tdf.sunflower.state.AddrUtil

interface Miner {
    fun start()
    fun stop()
    val address: Address
        get() = AddrUtil.empty()

    companion object {
        val NONE: Miner = object : Miner {
            override fun start() {}
            override fun stop() {}
        }
    }
}